src/librustc_mir_build/lints.rs

```diff
@@ -28,7 +28,12 @@ fn check_fn_for_unconditional_recursion<'tcx>(
     def_id: DefId,
 ) {
     let self_calls = find_blocks_calling_self(tcx, &body, def_id);
+
+    // Stores a list of `Span`s for every basic block. Those are the spans of `Call` terminators
+    // where we know that one of them will definitely be reached.
```

Comment:
If I understand right, when the vector is empty, it means that we **do not** always end up at a recursive call from this basic block. When it has one or more spans, it means we **do** always end up at a recursive call from this basic block, and the vector contains the spans of all the recursive calls we could possibly reach. Is this correct? If so, could you make this a bit clearer in the comment or by giving a better name to `results`?

Reply:
Yeah, that's right. I've updated the comment.